### ISSUE TITLE
[Shared] Config and Genesis refactoring improvement: Consider renaming `NodeModules`

### ISSUE THREAD

**Issue by deblasis** [pokt-network/pocket#292]
## Objective

Keep track and address areas of improvement. Specifically, in this issue we aim at addressing the comments below (which are related)

Related PR #235

## Origin Document

- https://github.com/pokt-network/pocket/pull/235#discussion_r986177118
- https://github.com/pokt-network/pocket/pull/235#discussion_r986192819
- https://github.com/pokt-network/pocket/pull/235#discussion_r986253946

## Goals

- [ ] Code quality

## Deliverable

- [ ] Understood and agreed upon implementation

## Non-goals / Non-deliverables

- Any change in logic

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
@deblasis Figured this could be done in parallel with the config refactoring.

**Comment by deblasis** [comment]
This has been addressed already and I forgot to close it. 

We discussed the concept of `NodeModule` in #235 (that I had introduced) but then in code-review we agreed it was unnecessary so I removed it.
